alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/DefaultAccessTokenSupplierTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.alloydb;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DefaultAccessTokenSupplierTest {
+  private final Instant now = Instant.now();
+  private final Instant past = now.plus(-1, ChronoUnit.HOURS);
+  private final Instant future = now.plus(1, ChronoUnit.HOURS);
+
+  private GoogleCredentials scopedCredentials;
+  private AtomicInteger refreshCounter;
+
+  @Before
+  public void setup() throws IOException {
+    refreshCounter = new AtomicInteger();
+
+    // Scoped credentials can't be refreshed.
+    scopedCredentials =
+        new GoogleCredentials(new AccessToken("my-scoped-token", null)) {
+          @Override
+          public AccessToken refreshAccessToken() throws IOException {
+            refreshCounter.incrementAndGet();
+            throw new IllegalStateException("Refresh not supported");
+          }
+        };
+  }
+
+  @Test
+  public void testEmptyTokenOnEmptyCredentials() throws IOException {
+    DefaultAccessTokenSupplier supplier = new DefaultAccessTokenSupplier(null);
+    assertThat(supplier.getTokenValue()).isEqualTo(null);
+  }
+
+  @Test
+  public void testWithValidToken() throws Exception {
+    // Google credentials can be refreshed
+    GoogleCredentials googleCredentials =
+        new GoogleCredentials(new AccessToken("my-token", Date.from(future))) {
+          @Override
+          public GoogleCredentials createScoped(String... scopes) {
+            return scopedCredentials;
+          }
+
+          @Override
+          public AccessToken refreshAccessToken() throws IOException {
+            refreshCounter.incrementAndGet();
+            return super.refreshAccessToken();
+          }
+        };
+
+    DefaultAccessTokenSupplier supplier =
+        new DefaultAccessTokenSupplier(new GoogleCredentialsFactory(googleCredentials));
+    Optional<AccessToken> token = supplier.get();
```

Comment:
This isn't part of the AccessTokenSupplier interface. Should we test through the public API instead?